includes/blocks/class-sensei-course-outline-block.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/**
+ * File containing the Sensei_Course_Outline_Block class.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Sensei_Course_Outline_Block
+ */
+class Sensei_Course_Outline_Block {
+	/**
+	 * Sensei_Course_Outline_Block constructor.
+	 */
+	public function __construct() {
+		add_action( 'enqueue_block_assets', [ $this, 'enqueue_assets' ] );
+		add_action( 'init', [ $this, 'register_course_template' ], 101 );
+		add_action( 'save_post_course', [ $this, 'save_course' ], 10, 3 );
+	}
+
+	/**
+	 * Enqueue frontend and editor assets.
+	 *
+	 * @access private
+	 */
+	public function enqueue_assets() {
+		if ( 'course' !== get_post_type() ) {
+			return;
+		}
+
+		Sensei()->assets->enqueue( 'sensei-course-outline-script', 'blocks/course-outline/index.js' );
+		Sensei()->assets->enqueue( 'sensei-course-outline-style', 'blocks/course-outline/style.css' );
+	}
+
+	/**
+	 * Register course template.
+	 *
+	 * @access private
+	 */
+	public function register_course_template() {
+		$post_type_object = get_post_type_object( 'course' );
+
+		$post_type_object->template = [
+			[ 'sensei-lms/course-outline' ],
+		];
+	}
+
+	/**
+	 * Save course.
+	 *
+	 * @param int      $post_id Post ID.
+	 * @param \WP_Post $post    The post.
+	 * @param bool     $update  Whether this is an existing post being updated.
+	 */
+	public function save_course( $post_id, $post, $update ) {
+		$content = $post->post_content;
+		$blocks  = parse_blocks( $content );
+
+		// TODO: Get the attributes and save the data.
```

Comment:
Here's an approach to save the data. We can send the attributes to the server and save it. It's the approach the Crowdsignal is using and it looks a safe way to do it.

Reply:
While taking a shower I had an insight into this. 😜🚿

The block will check if our attributes are valid when opening the block, but I think if we use the same [approach that we used in the POCs](https://github.com/Automattic/sensei/pull/3508/files#diff-fa727fcea369ce55374615c28f66b53aR83-R101), it'll not be a problem because we can make the first render with the current attributes, updating it immediately after.

Reply:
I don't think this needs to be in this PR. Hooking into the save function (either in PHP or JS) is a bit dangerous as it could be called multiple times, and we need to create new lessons & modules here based on the changes. 

The best candidate for all that is probably a JS store that keeps proper track of any newly created lessons and such, and handles the API calls.

Reply:
I reverted it for now, as you suggested. We can dig into that in future PRs. 🙂